app/controllers/api/automate_domains_controller.rb

```diff
@@ -27,6 +27,10 @@ def refresh_from_source_resource(type, id = nil, data = nil)
       end
     end
 
+    def fetch_automate_domains_href_slug(resource)
```

Comment:
Does this mean this method gets invoked via some meta programming? Seems like this could be better implemented using polymorphism. But definitely not as part of this PR ;)